pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/ddp/tissue/TissueSMIDDao.java

```diff
@@ -48,15 +44,17 @@ public class TissueSMIDDao {
                     + "LEFT JOIN ddp_participant as p ON (p.participant_id = ins.participant_id) "
                     + "LEFT JOIN ddp_instance as realm ON (realm.ddp_instance_id = p.ddp_instance_id) "
                     + "WHERE realm.instance_name = ?";
+    private static final Logger logger = LoggerFactory.getLogger(TissueSMIDDao.class);
 
-    public String getTypeForName(String type) {
+    public static List<String> getSmIdPksForTissue(String tissueId) {
+        List<String> smIds = new ArrayList<>();
         SimpleResult results = inTransaction((conn) -> {
             SimpleResult dbVals = new SimpleResult();
-            try (PreparedStatement stmt = conn.prepareStatement(SQL_TYPE_ID_FOR_TYPE)) {
-                stmt.setString(1, type);
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_SM_ID_BASED_ON_TISSUE_ID)) {
+                stmt.setString(1, tissueId);
                 try (ResultSet rs = stmt.executeQuery()) {
-                    if (rs.next()) {
-                        dbVals.resultValue = rs.getString(DBConstants.SM_ID_TYPE_ID);
+                    while (rs.next()) {
+                        smIds.add(rs.getString(DBConstants.SM_ID_PK));
```

Comment:
This method finds the sm ids that all belong to the same tissue by the tissue id. The idea is if that tissue gets deleted, the sm ids belonging to it should also get deleted